### PR TITLE
docs: debugging docs

### PR DESCRIPTION
**Pull request by gkartalis**
This PR resolves [PHIRE-1348] 

### Description

Updates debugging docs to point to react native docs.

Also adds commands in `package.json`:

`yarn react-devtools` and `yarn start-dev` to access the new rn tools.

### PR Checklist

- [ ] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- update debugging docs 

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[PHIRE-1348]: https://artsyproduct.atlassian.net/browse/PHIRE-1348?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Dev changes (update debugging docs - gkartalis)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 50ecede88458f0eb050b7e63713ef200b7c6746e
</p>

File: package.json [hunk 1]
```diff
@@ -72,6 +72,8 @@
     "setup:releases": "./scripts/setup/setup-env-for-artsy && ./scripts/setup/setup-env-for-releases",
     "setup:releases:update!": "scripts/setup/update-env-for-releases",
     "start": "concurrently 'yarn relay:watch' 'react-native start'",
+    "start-dev": "npx react-native start --experimental-debugger",
```
[the remark this answers is not in the document]
Comment (gkartalis) [reply package.json:75]: fair point, i did tried doing that though and seems like the concurrently command just ignores the `--experimental-debugger` part unfortunately..

Also tried `yarn start --experimental-debugger` after adding npx in yarn start and also didn't work

**Review by MrSltun — APPROVED** [review]

File: package.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (gkartalis) [reply package.json:75]: 👀 did you read the comments above? 😆

[the remark this answers is not in the document]
Comment (damassi) [reply package.json:75]: All good!

**Pull request merged**